### ISSUE TITLE
Loss of compatibility on v3.2.1.26

### ISSUE THREAD

**Issue by enolgargon** [Kirollos/DiscordEuroscope#8]
The plugin is only working up to version r25. With version r25 you can load the plugin and use it without problems, but when upgrading to version r26 Euroscope crashes when opening due to the plugin. If you install version r26 (without the plugin) and open Euroscope, it works correctly. When loading the plugin a problem occurs and the program crashes without showing any error message.

**Comment by Kirollos** [comment]
Sorry for being late, I will investigate and update you.

**Comment by Kirollos** [comment]
@enolgargon Can you confirm if the same issue persists on r29?

**Comment by enolgargon** [comment]
Yes. I found when I upgrade from the r25 to the r29. I thought that it could be a problem with the r29 version because it appears with the text "It might be less stable than the versions before" on the official website. So I downgrade my version until it works. To prevent this testing process from affecting the Euroscope, I did a clean install of the Euroscope on both the r25 and r26 versions. The new installation with r25 worked and the one with r26 did not.

**Comment by amosngSP** [comment]
Hey, just wanted to follow this issue as well, a fellow controller had the same issue where upon loading the profile, it will crash Euroscope straight away. Just here to follow up on this issue as well, seems like only 1 or 2 controllers had this issue.

**Comment by Kirollos** [comment]
> Hey, just wanted to follow this issue as well, a fellow controller had the same issue where upon loading the profile, it will crash Euroscope straight away. Just here to follow up on this issue as well, seems like only 1 or 2 controllers had this issue.

@amosngSP Hi, which ES version?

**Comment by Kirollos** [comment]
The weird thing is that I am able to run the plugin fine on both r26 and r29.

**Comment by Hollywood0001** [comment]
Can confirm it does not work correctly in r26. Out of the last 15/20ish sessions online, I've only seen DRP populate once.

**Comment by Kirollos** [comment]
> Can confirm it does not work correctly in r26. Out of the last 15/20ish sessions online, I've only seen DRP populate once.

Ok, I will recompile with latest VS and see what happens. I will be giving you a dll to test in few hours today.

**Comment by Kirollos** [comment]
@enolgargon 
@amosngSP 
@Hollywood0001 
Dear, Can you please try the attached DLL and report to me? Thanks!
[DiscordEuroscope.zip](https://github.com/Kirollos/DiscordEuroscope/files/6992847/DiscordEuroscope.zip)

**Comment by Hollywood0001** [comment]
@Kirollos Just tried it both as an observer and as ground (made sure both callsigns were in the .txt), discord did not pick up the rich presence.

**Comment by Kirollos** [comment]
> @Kirollos Just tried it both as an observer and as ground (made sure both callsigns were in the .txt), discord did not pick up the rich presence.

But it loads without error, correct? Can you confirm that you receive a DiscordEuroscope message in "Message" window? Also if you are running discord as administrator, you should also run ES as admin.

**Comment by Hollywood0001** [comment]
> > @Kirollos Just tried it both as an observer and as ground (made sure both callsigns were in the .txt), discord did not pick up the rich presence.
> 
> But it loads without error, correct? Can you confirm that you receive a DiscordEuroscope message in "Message" window? Also if you are running discord as administrator, you should also run ES as admin.

Yes, I did see the message about it having parsed 5 callsigns. I thought I was running ES as administrator already, but apparently not. Restarted it as administrator, and Rich Presence started working immediately, thank you!

**Comment by Kirollos** [comment]
> > > @Kirollos Just tried it both as an observer and as ground (made sure both callsigns were in the .txt), discord did not pick up the rich presence.
> > 
> > 
> > But it loads without error, correct? Can you confirm that you receive a DiscordEuroscope message in "Message" window? Also if you are running discord as administrator, you should also run ES as admin.
> 
> Yes, I did see the message about it having parsed 5 callsigns. I thought I was running ES as administrator already, but apparently not. Restarted it as administrator, and Rich Presence started working immediately, thank you!

Thank you!
I will keep this thread up for a few days if anyone is still having issues, please let me know.

**Comment by amosngSP** [comment]
Hi @Kirollos sorry for this late reply.

I've sent the plugins to the controllers who are affected by this issue. Waiting to get their feedback on this.

**Comment by amosngSP** [comment]
Hi, my controller still has the issue of Euroscope crashing on him even after downloading the version you provided.
`Faulting application name: EuroScope.exe, version: 3.2.1.26, time stamp: 0x601fc229
Faulting module name: ucrtbase.dll, version: 10.0.19041.789, time stamp: 0x82dc99a2
Exception code: 0xc0000409
Fault offset: 0x0009d132
Faulting process ID: 0x954
Faulting application start time: 0x01d7950f2544714b
Faulting application path: C:\Program Files (x86)\EuroScope\EuroScope.exe
Faulting module path: C:\Windows\System32\ucrtbase.dll
Report ID: b3e9a1de-3cd0-4169-8933-5eeb5017ccfc
Faulting package full name: 
Faulting package-relative application ID: `

**Comment by Kirollos** [comment]
@amosngSP Hi, can you make sure the controller has this installed?
https://aka.ms/vs/16/release/vc_redist.x86.exe
Thanks!
